error_check_logic.py

```diff
@@ -12,229 +13,268 @@
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class student groups against Student Groups: ' + result
-  result = CheckStudentGroupsUsed(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck if all Student Groups are used: ' + result
-  result = CheckClassEmails(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class student emails against Students: ' + result
-  result = CheckCurrentGrade(institution, session)
-  if result != 'OK':
-    setup_msg = 'FAIL'
-  error_chk += '\n\nCheck Class current_grade against Student current_grade: ' + result
-  return setup_msg, error_chk
+class Checker(object):
+  def __init__(self):
+    self.error_check_msg = 'OK' # setup_msg
+    self.error_check_detail = '' # error_chk
+
+  # Modules that change admin setup data should call this
+  # with status = 'UNKNOWN'.
+  @classmethod
+  def setStatus(self, institution, session, status = 'UNKNOWN'):
+    models.ErrorCheck.Store(institution, session, status)
+
+  @classmethod
+  def getStatus(self, institution, session):
+    return models.ErrorCheck.Fetch(institution, session)
+
+  # Returns two values:
+  #   error_check_msg: 'OK' if all tests pass
+  #                    'FAIL' if any single test fails
+  #   error_check_detail: String containing detailed results
+  #                       for ALL tests.
+  # Sets datastore to value of error_check_msg
+  def Run(self, institution, session):
+    dayparts = models.Dayparts.Fetch(institution, session)
+    classes = models.Classes.Fetch(institution, session)
+    students = models.Students.Fetch(institution, session)
+    requirements = models.Requirements.Fetch(institution, session)
+    class_groups = models.GroupsClasses.Fetch(institution, session)
+    student_groups = models.GroupsStudents.Fetch(institution, session)
+
+    valid_dayparts = schemas.dayparts.IsValid(dayparts)
+    valid_classes = schemas.classes.IsValid(classes)
+    valid_students = schemas.students.IsValid(students)
+    valid_requirements = schemas.requirements.IsValid(requirements)
+    valid_class_groups = schemas.class_groups.IsValid(class_groups)
+    valid_student_groups = schemas.student_groups.IsValid(student_groups)
+
+    self.error_check_detail += '\nValidate Dayparts: ' +\
+      self._Validate(dayparts, valid_dayparts)
+    self.error_check_detail += '\nValidate Classes: ' +\
+      self._Validate(classes, valid_classes)
+    self.error_check_detail += '\nValidate Students: ' +\
+      self._Validate(students, valid_students)
+    self.error_check_detail += '\nValidate Requirements: ' +\
+      self._Validate(requirements, valid_requirements)
+    self.error_check_detail += '\nValidate Class Groups: ' +\
+      self._Validate(class_groups, valid_class_groups)
+    self.error_check_detail += '\nValidate Student Groups: ' +\
+      self._Validate(student_groups, valid_student_groups)
```

Comment:
It might make sense to put a try catch block around the following code.  If it's already invalid after this point then the subsequent code might throw exceptions. If that happens we wouldn't see any of the above errors because the page might fail to load.